datastore/src/test/java/io/spine/server/storage/datastore/DsAggregateStorageShould.java

```diff
@@ -98,6 +112,67 @@ public void fail_to_write_invalid_record() {
                             AggregateEventRecord.getDefaultInstance());
     }
 
+    @Test
+    public void set_limit_for_history_backward_query() {
+        final DsAggregateStorage<ProjectId> storage = (DsAggregateStorage<ProjectId>) getStorage();
+        final int batchSize = 10;
+        final AggregateReadRequest<ProjectId> request = new AggregateReadRequest<>(newId(),
+                                                                                   batchSize);
+        final EntityQuery historyBackwardQuery = storage.historyBackwardQuery(request);
+
+        final int queryLimit = historyBackwardQuery.getLimit();
+        assertEquals(batchSize, queryLimit);
+    }
+
+    @Test
+    public void allow_aggregate_loading_if_snapshot_trigger_was_decreased_in_AggregateRepository() {
+        final ProjectAggregateRepository repository = new ProjectAggregateRepository();
+        register(repository);
+
+        final ProjectId id = newId();
+        final int initialSnapshotTrigger = 10;
+
+        // To restore an aggregate using a snapshot and events.
+        final int tasksCount = initialSnapshotTrigger * 2 - 1;
+
+        repository.setSnapshotTrigger(initialSnapshotTrigger);
+        for (int i = 0; i < tasksCount; i++) {
+            repository.dispatch(env(addTask(id)));
+        }
+
+        final int minimalSnapshotTrigger = 1;
+        repository.setSnapshotTrigger(minimalSnapshotTrigger);
+        final ProjectAggregate aggregate = repository.find(id)
+                                                     .get();
+        assertEquals(tasksCount, aggregate.getState()
+                                          .getTaskCount());
+    }
+
+    private static void register(AggregateRepository repository) {
```

Comment:
Let's move things like this one to `Given` (similar to what we do in `core-java`). This method is both excessive in the test code and weird — as it's `static`, looks like a registration, but in fact creates a bounded context each time.

Let's discuss if needed. But at the moment this piece of code is difficult to reuse properly.